### ISSUE TITLE
Wrong hparams when using lightning cli

### ISSUE THREAD

**Issue by BakerBunker** [Lightning-AI/pytorch-lightning#20118]
### Bug description

When using lightning 2.3.3 and lightning cli, the `hparams` property becomes
```
{
"_instantiator": ...,
"class_path": ...,
"init_args": ...,
}
```
It is correct when using lightning 2.2.5

### What version are you seeing the problem on?

2.3.3

### How to reproduce the bug

```python
instantiate a `LightningModule` with `self.save_hyperparameters()` using cli and `print(self.hparams)`
```

### Error messages and logs

N/A

### Environment

<details>
  <summary>Current environment</summary>

```
#- PyTorch Lightning Version (e.g., 1.5.0):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
```

</details>

cc @carmocca @mauvilsa

**Comment by awaelchli** [comment]
Does this PR fix this? #20068

**Comment by mauvilsa** [comment]
The instructions to reproduce are not very clear. How do you instantiate? With 2.3 the idea is that now it is possible to instantiate using `Module.load_from_checkpoint`, but I guess this is not how you are instantiating. Or is it?

**Comment by BakerBunker** [comment]
> Hi @BakerBunker Does this PR fix this? #20068

Yes, the same issue as https://github.com/Lightning-AI/pytorch-lightning/issues/19977 , thanks!
